Mathlib/CategoryTheory/Preadditive/Injective.lean

```diff
@@ -346,8 +346,40 @@ def mapInjectivePresentation (adj : F ⊣ G) [F.PreservesMonomorphisms] (X : D)
     haveI : PreservesLimitsOfSize.{0, 0} G := adj.rightAdjointPreservesLimits; infer_instance
 #align category_theory.adjunction.map_injective_presentation CategoryTheory.Adjunction.mapInjectivePresentation
 
+/-- Given an adjunction `F ⊣ G` such that `F` preserves finite limits and is faithful,
+  then any injective presentation of `F(X)` can be pulled back to an injective presentation of `X`.
+  This is similar to `mapInjectivePresentation`. -/
+def injectivePresentationOfMapInjectivePresentation (adj : F ⊣ G)
+    [PreservesFiniteLimits F] [Faithful F] (X : C)
+    (I : InjectivePresentation <| F.obj X) :
+    InjectivePresentation X where
+  J := G.obj I.J
+  injective := Injective.injective_of_adjoint adj _
+  f := adj.homEquiv _ _ I.f
+  mono := by
+    have eq1 : F.map (adj.homEquiv _ _ <| I.f) ≫ (adj.counit.app _) = I.f := by simp
+    have : Mono (F.map (adj.homEquiv _ _ <| I.f) ≫ (adj.counit.app _)) := eq1.symm ▸ I.mono
+    have : Mono (F.map (adj.homEquiv _ _ <| I.f)) := mono_of_mono _ (adj.counit.app I.J)
+    simp_all only [Functor.id_obj, Functor.mono_map_iff_mono]
```

Comment:
```suggestion
```

You may remove the proof of `mono` if you add this instance:
```
instance (adj : F ⊣ G) {X : C} {Y : D} (f : F.obj X ⟶ Y) [Mono f] [Faithful F] :
    Mono (adj.homEquiv _ _ f) where
  right_cancellation _ _ h := F.map_injective
    (by simpa only [← cancel_mono f] using (adj.homEquiv _ _).injective (by simpa using h))
```

Reply:
thanks